library/std/src/sys/unix/memchr.rs

```diff
@@ -26,7 +26,7 @@ pub fn memrchr(needle: u8, haystack: &[u8]) -> Option<usize> {
                 haystack.len(),
             )
         };
-        if p.is_null() { None } else { Some(p as usize - (haystack.as_ptr() as usize)) }
+        if p.is_null() { None } else { Some(p.addr() - haystack.as_ptr().addr()) }
     }
```

Comment:
Definitely not touching a memchr implementation with a 10-foot pole

Reply:
Same thing here wrt a comment like:
```rust
// FIXME: this should *likely* use `offset_from`, but more
// investigation is needed (including running tests in miri).
```